source/cfa.h

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2015-2016 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SPVTOOLS_CFA_H_
+#define SPVTOOLS_CFA_H
+
+#include <algorithm>
+#include <cassert>
+#include <functional>
+#include <map>
+#include <unordered_map>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+using std::find;
+using std::function;
+using std::get;
+using std::pair;
+using std::unordered_map;
+using std::unordered_set;
+using std::vector;
+
+namespace spvtools {
+
+template<class BB> class CFA {
```

Comment:
This class deserves a comment, especially since its name is an acronym.  BB also deserves a mention as "basic block".